common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -179,12 +179,26 @@ public static StringSearch getStringSearch(
       final UTF8String left,
       final UTF8String right,
       final int collationId) {
+
+    if (collationId == UTF8_BINARY_LCASE_COLLATION_ID) {
```

Comment:
what if the collationId is zero?

Reply:
That is special case and it is tested explicitly in each method so we can call already existing methods for binary comparison.